### PR TITLE
fix istioctl precheck unused code logic

### PR DESCRIPTION
**Pull request by monsteredc**
Fix issue #35451.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: istioctl/cmd/precheck.go [hunk 1]
```diff
@@ -308,7 +308,10 @@ func checkListeners(cli kube.ExtendedClient, namespace string) (diag.Messages, e
 					continue
 				}
 				ip := net.ParseIP(bind)
-				ip.IsGlobalUnicast()
+				// skip the global unicast address
```
Comment (howardjohn) [inline istioctl/cmd/precheck.go:311]: Why? I think we should preserve the old behavior if the issue we are fixing is just "deadcode" and remove the line?

Comment (monsteredc) [reply istioctl/cmd/precheck.go]: Alright, I thought checking the `IsGlobalUnicast` condition was one of the options to fix the issue after looking into the code logic in the code block. Just revised to a newer version. Thanks!

**Review by howardjohn — APPROVED** [review]

**Comment by monsteredc** [comment]
@shamsher31 PTAL.

**Review by shamsher31 — APPROVED** [review]

**Pull request merged**